functionaltests/src/test/java/specs/multitenancy/BaseTenant2Fixture.java

```diff
@@ -0,0 +1,131 @@
+package specs.multitenancy;
+
+import com.commercetools.pspadapter.payone.domain.ctp.TypeCacheLoader;
+import com.google.common.base.Preconditions;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.LoadingCache;
+import io.sphere.sdk.client.BlockingSphereClient;
+import io.sphere.sdk.client.SphereClientFactory;
+import io.sphere.sdk.payments.Payment;
+import io.sphere.sdk.payments.queries.PaymentByIdGet;
+import io.sphere.sdk.types.Type;
+import org.junit.BeforeClass;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import specs.BaseFixture;
+import specs.paymentmethods.BaseNotifiablePaymentFixture;
+
+import java.net.MalformedURLException;
+
+/**
+ * Base class for second tenant payments tests. It overrides CTP client, type cache map, own tenant name.
+ */
+abstract public class BaseTenant2Fixture extends BaseNotifiablePaymentFixture {
+
+    private static Logger LOG = LoggerFactory.getLogger(BaseTenant2Fixture.class);
+
+    protected static final String TEST_DATA_TENANT_NAME_2 = "TEST_DATA_TENANT_NAME_2";
+    protected static final String TEST_DATA_CT_PROJECT_KEY_2 = "TEST_DATA_CT_PROJECT_KEY_2";
+    protected static final String TEST_DATA_CT_CLIENT_ID_2 = "TEST_DATA_CT_CLIENT_ID_2";
+    protected static final String TEST_DATA_CT_CLIENT_SECRET_2 = "TEST_DATA_CT_CLIENT_SECRET_2";
+
+    protected static final String TEST_DATA_PAYONE_MERCHANT_ID_2 = "TEST_DATA_PAYONE_MERCHANT_ID_2";
+    protected static final String TEST_DATA_PAYONE_SUBACC_ID_2 = "TEST_DATA_PAYONE_SUBACC_ID_2";
+    protected static final String TEST_DATA_PAYONE_PORTAL_ID_2 = "TEST_DATA_PAYONE_PORTAL_ID_2";
+    protected static final String TEST_DATA_PAYONE_KEY_2 = "TEST_DATA_PAYONE_KEY_2";
+
+    private static BlockingSphereClient ctpClient2;
+
+    private static LoadingCache<String, Type> typeCache2;
```

Comment:
It is legacy implementation, i'm not 100% sure, but guess to encapsulate loading/caching values, so you don't care about this calls, you just do [`getTypeCache().get(typeName).getId()`](https://github.com/commercetools/commercetools-payone-integration/blob/55d099c951730a8f0fbebc7b7fe4614e53931597/functionaltests/src/test/java/specs/BaseFixture.java#L457-L457) and the _LoadingCache_ class makes necessary load-cache-return steps on demand.